src/test/java/org/breedinginsight/services/parsers/TraitFileParserUnitTest.java

```diff
@@ -122,6 +121,84 @@ void parseExcelEmptyFormulaColumnNames() {
         assertThrows(ParsingException.class, () -> parser.parseExcel(inputStream), "expected parsing exception");
     }
 
+    @Test
+    @SneakyThrows
+    void parseCsvActiveColumnInvalidValue() {
+        File file = new File("src/test/resources/files/data_one_row_invalid_active_value.csv");
+        InputStream inputStream = new FileInputStream(file);
+
+        assertThrows(ParsingException.class, () -> parser.parseCsv(inputStream), "expected parsing exception");
```

Comment:
This goes for other parsing exception tests, should there be a validation of the message of the exception?  For this test, the exception message should be "Invalid trait status value"

Reply:
In general I try to avoid checking messages on exceptions because usually it's just added maintenance cost of tests but in this case since the same exception is being used a lot in different contexts I agree that it would be valuable. It is currently possible that a test would pass due to an exception occurring somewhere else in the code than expected in the test case so I will add exception message checks for the unit tests. I can also add some constants for the exception messages to use in the parser and unit tests.

Reply:
@timparsons and I talked about this. I realized after talking that I was already far sown the path of using an enum to define the types and it would be faster to just finish off the rudimentary version of that. Hope you're good with that Tim. I'll make a card to make it nicer using interfaces. Pushed changes.

Reply:
Added card: https://trello.com/c/axsfTss0